### ISSUE TITLE
🐛 Update parser to support `satisfies` keyword (Typescript 4.9+)

### ISSUE THREAD

**Issue by lildeadprince** [rome/tools#3748]
### Environment information

```block
CLI:
  Version:              10.0.1
  Color support:        true

Platform:
  CPU Architecture:     x86_64
  OS:                   windows

Environment:
  ROME_LOG_DIR:         unset
  NO_COLOR:             unset
  TERM:                 "cygwin"

Rome Configuration:
  Status:               loaded
  Formatter disabled:   false
  Linter disabled:      true

Workspace:
  Open Documents:       0

Discovering running Rome servers...

Server:
  Status:               stopped
```

### What happened?

Today ([Nov 15 2022](https://github.com/microsoft/TypeScript/issues/50457)) Typescript 4.9 reaches planned release date. The brand new [`satisfies`](https://devblogs.microsoft.com/typescript/announcing-typescript-4-9-beta/#the-satisfies-operator) keyword is coming into play.

The updates for that keyword has already been included before in other tools like esbuild (since 0.5.14) and SWC (since 1.3.15).

Rome tools right now fail [(playground link)](https://docs.rome.tools/playground/?code=dAB5AHAAZQAgAEMAbwBsAG8AcgBzACAAPQAgACIAcgBlAGQAIgAgAHwAIAAiAGcAcgBlAGUAbgAiACAAfAAgACIAYgBsAHUAZQAiADsACgAKAHQAeQBwAGUAIABSAEcAQgAgAD0AIABbAHIAZQBkADoAIABuAHUAbQBiAGUAcgAsACAAZwByAGUAZQBuADoAIABuAHUAbQBiAGUAcgAsACAAYgBsAHUAZQA6ACAAbgB1AG0AYgBlAHIAXQA7AAoACgBjAG8AbgBzAHQAIABwAGEAbABlAHQAdABlACAAPQAgAHsACgAgACAAIAAgAHIAZQBkADoAIABbADIANQA1ACwAIAAwACwAIAAwAF0ALAAKACAAIAAgACAAZwByAGUAZQBuADoAIAAiACMAMAAwAGYAZgAwADAAIgAsAAoAIAAgACAAIABiAGwAZQB1ADoAIABbADAALAAgADAALAAgADIANQA1AF0ACgAvAC8AIAAgAH4AfgB%2BAH4AIABUAGgAZQAgAHQAeQBwAG8AIABpAHMAIABuAG8AdwAgAGMAYQB1AGcAaAB0ACEACgB9ACAAcwBhAHQAaQBzAGYAaQBlAHMAIABSAGUAYwBvAHIAZAA8AEMAbwBsAG8AcgBzACwAIABzAHQAcgBpAG4AZwAgAHwAIABSAEcAQgA%2BADsA&typescript=true&jsx=false) to parse code containing such keyword:

<details>
  <summary>Error log</summary>

```
> rome format ./src --write

./index.ts:12:3 parse ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  × Expected a semicolon or an implicit semicolon after a statement, but found none

    10 │   blue: [0, 0, 255],
    11 │ //  ~~~~ The typo is now caught!
  > 12 │ } satisfies Record<Colors, string | RGB>;
       │   ^^^^^^^^^
    13 │ 

  i An explicit or implicit semicolon is expected here...

    10 │   blue: [0, 0, 255],
    11 │ //  ~~~~ The typo is now caught!
  > 12 │ } satisfies Record<Colors, string | RGB>;
       │   ^^^^^^^^^
    13 │ 

  i ...Which is required to end this statement

     5 │ type RGB = [red: number, green: number, blue: number];
     6 │ 
   > 7 │ const palette = {
       │ ^^^^^^^^^^^^^^^^^
   > 8 │   red: [255, 0, 0],
   > 9 │   green: '#00ff00',
  > 10 │   blue: [0, 0, 255],
  > 11 │ //  ~~~~ The typo is now caught!
  > 12 │ } satisfies Record<Colors, string | RGB>;
       │ ^^
    13 │ 


```

</details>

### Expected result

`satisfies` keyword is expected to be parsed correctly.

_I guess it should not be very complex, but unfortunately I'm not familiar enough with Rust. Furthermore, it is code related to the parser core (and files [generated](https://github.com/rome/tools/blob/main/crates/rome_js_syntax/src/generated/kind.rs#L607) from I have no idea what source).
Therefore I won't try fixing it and filing a PR myself_ 

### Code of Conduct

- [X] I agree to follow Rome's Code of Conduct

**Comment by nissy-dev** [comment]
I'll try to take this issue!

**Comment by MichaReiser** [comment]
> I'll try to take this issue!

Awesome! Let me know if you need any help. You may want to comment here with your proposed AST changes if they become more involved.

**Comment by ematipico** [comment]
@MichaReiser I'd like your opinion: does it make sense to work on new features released in version 4.9 while we haven't implemented changes for version 4.8?

I know we have an issue to track 4.8 parser changes

**Comment by MichaReiser** [comment]
> @MichaReiser I'd like your opinion: does it make sense to work on new features released in version 4.9 while we haven't implemented changes for version 4.8?
> 
> I know we have an issue to track 4.8 parser changes

I don't think it's an issue, for as long as there are no dependencies and satisfies is probably one of the more important features.
